src/codegen/function.rs

```diff
@@ -224,19 +224,32 @@ pub fn declaration(env: &Env, analysis: &analysis::functions::Info) -> String {
     )
 }
 
-pub fn declaration_futures(env: &Env, analysis: &analysis::functions::Info) -> String {
+pub fn declaration_futures(
+    env: &Env,
+    analysis: &analysis::functions::Info,
+    in_trait: bool,
+) -> String {
     let async_future = analysis.async_future.as_ref().unwrap();
 
     let return_str = if let Some(ref error_parameters) = async_future.error_parameters {
-        format!(
-            " -> Pin<Box_<dyn std::future::Future<Output = Result<{}, {}>> + 'static>>",
-            async_future.success_parameters, error_parameters
-        )
-    } else {
+        if in_trait {
```

Comment:
https://github.com/gtk-rs/gtk4-rs/pull/1421#discussion_r1257719379 FWIW. This will now create differently behaving API depending on whether it's in a trait or not